### ISSUE TITLE
[Feat] Add padding option to defaultBounds

### ISSUE THREAD

**Issue by francescocretti** [visgl/react-google-maps#391]
### Target Use Case

Initialise the Map instance with default bounds and a padding value in pixels, without the need to call `fitBounds` right after initialising the map to apply padding.

### Proposal

The `defaultBounds` is very useful in several use cases to initialise a map with a specific view previously calculated from some incoming data (or any other source).
The bounds approach is also very convenient when it comes to padding: the Google Maps API accepts a padding in pixels as an argument of the `fitBounds` method ([docs here](https://developers.google.com/maps/documentation/javascript/reference/map#Map.fitBounds)).

Since defining the `defaultBounds` prop leads to invoking the `fitBounds` method under the hood (https://github.com/visgl/react-google-maps/blob/39603438948c285e8a14e7e4e8d01034b58e1db4/src/components/map/use-map-instance.ts#L142), it would be very handy to add the chance to pass also a padding option. 

Could be something like that:

```javascript
 <Map
   defaultBounds={bounds}
   defaultBoundsPadding={{
    left: 10,
    right: 10,
    top: 10,
    bottom: 10,
   }}
/>
```

It shouldn't be a big change, I'm available to help if needed.
Thanks a lot.

**Comment by usefulthink** [comment]
I like this, in fact I haven't thought about that padding parameter. 

Maybe instead of adding an additional prop for that, we could add an optional field `padding` to the `defaultBounds` prop.

**Comment by francescocretti** [comment]
@usefulthink sure, I don't have any preferences!
Using of an additional prop can be easier to handle and also guarantees backwards compatibility (since if u don't need padding you only pass `defaultBounds` as it is now).

Let me know what u prefer and I can open a PR if you'd like :)

**Comment by usefulthink** [comment]
I think specifying the padding with the same prop you're using to specify the bounds makes sense, with padding being optional it also should be backwards-compatible. We could just extend the type in https://github.com/visgl/react-google-maps/blob/39603438948c285e8a14e7e4e8d01034b58e1db4/src/components/map/index.tsx#L82 to `google.maps.LatLngBoundsLiteral & {padding ?: number | google.maps.Padding}`.

**Comment by francescocretti** [comment]
Okay, let's do it ;)

**Comment by usefulthink** [comment]
This will be released with [v1.1.0](#395), probably around next week.

**Comment by karldanninger** [comment]
I think this isn't working anymore.

```
const offset = { left: 40, right: 40, bottom: 40, top: 40 };
...
defaultBounds: { ...bounds, padding: { ...offset } }
```

effectively ignores the padding
